openassessment/xblock/grade_mixin.py

```diff
@@ -266,32 +252,189 @@ def _rubric_criteria_grade_context(self, peer_assessments, self_assessment):
+            criterion['assessments'] = self._assessment_grades(
+                submission_uuid, criterion,
+                staff_assessment=staff_assessment,
+                peer_assessments=peer_assessments,
+                example_based_assessment=example_based_assessment,
+                self_assessment=self_assessment,
+            )
+
+            # Record whether there is any feedback provided in the assessments
+            criterion['has_feedback'] = has_feedback(criterion['assessments'])
+
+            # Although we prevent course authors from modifying criteria post-release,
+            # it's still possible for assessments created by course staff to
+            # have criteria that differ from the current problem definition.
+            # It's also possible to circumvent the post-release restriction
+            # if course authors directly import a course into Studio.
+            # If this happens, we simply leave the score blank so that the grade
+            # section can render without error.
+            criterion['median_score'] = median_scores.get(criterion_name, '')
+            criterion['total_value'] = max_scores.get(criterion_name, '')
+
+        return {
+            'criteria': criteria,
+            'additional_feedback': self._additional_feedback(
+                staff_assessment=staff_assessment,
+                peer_assessments=peer_assessments,
+                self_assessment= self_assessment,
+            ),
+        }
+
+    def _assessment_grades(
+            self, submission_uuid, criterion, staff_assessment, peer_assessments,
+            example_based_assessment, self_assessment
+    ):
+        """
+        Returns an array of assessments with their associated grades.
+        """
+        def _get_assessment_part(title, part_criterion_name, assessment):
+            """
+            Returns the assessment part for the given criterion name.
+            """
+            if assessment:
+                for part in assessment['parts']:
+                    if part['criterion']['name'] == part_criterion_name:
+                        part['title'] = title
+                        return part
+            return None
+
+        criterion_name = criterion['name']
+        staff_assessment_part = _get_assessment_part(_('Staff Grade'), criterion_name, staff_assessment)
+        peer_assessment_part = {
+            'title': _('Peer Median Grade'),
+            'criterion': criterion,
+            'option': self._peer_median_option(submission_uuid, criterion),
+            'individual_assessments': [
+                _get_assessment_part(
+                    _('Peer {peer_index}').format(peer_index=index + 1),
+                    criterion_name,
+                    peer_assessment
+                )
+                for index, peer_assessment in enumerate(peer_assessments)
+                ],
+        } if len(peer_assessments) > 0 else None
```

Comment:
Yikes, I can't read this code. Can you change this around so it is more readable?